stdlib/public/core/StringBridge.swift

```diff
@@ -337,42 +360,59 @@ internal func _cocoaASCIIPointer(_ str: _CocoaString) -> UnsafePointer<UInt8>? {
   return _NSStringASCIIPointer(_objc(str))
 }
 
-private enum CocoaStringPointer {
-  case ascii(UnsafePointer<UInt8>)
-  case utf8(UnsafePointer<UInt8>)
-  case utf16(UnsafePointer<UInt16>)
-  case none
-}
-
-@_effects(readonly)
-private func _getCocoaStringPointer(
-  _ cfImmutableValue: _CocoaString
-) -> CocoaStringPointer {
-  if let asciiPtr = _cocoaASCIIPointer(cfImmutableValue) {
-    // NOTE: CFStringGetCStringPointer means ASCII
-    return .ascii(asciiPtr)
+// This does not create a bridged String that's safe to use past the current
+// stack frame, e.g. it does not defensively copy it and so on.
+@_effects(releasenone)
+internal func _withTemporaryBridgedCocoaString<R>(
+  _ cocoaString: _CocoaString?,
+  _ work: (String) throws -> R
+) rethrows -> R {
+  guard let cocoaString = cocoaString else {
+    return try work("")
   }
-  if let utf16Ptr = _stdlib_binary_CFStringGetCharactersPtr(cfImmutableValue) {
-    return .utf16(utf16Ptr)
+  switch _KnownCocoaString(cocoaString) {
+  case .storage(let storage):
+    return try work(storage.asString)
+  case .shared(let storage):
+    return try work(storage.asString)
+    #if !(arch(i386) || arch(arm))
+  case .tagged(let str):
+    let tmpStr = String(_StringGuts(_SmallString(taggedCocoa: str)))
+    return try work(tmpStr)
+    #endif
+  case .mutable(let storage):
+    return try work(storage.asString)
+  case .cocoa(let str):
+    let length = _stdlib_binary_CFStringGetLength(str)
```

Comment:
Probably add a comment that we'll try to form an immortal contiguous-ASCII string if we can, otherwise we have to fall back to a lazily bridged object.